prometheus/registry.go

```diff
@@ -505,22 +470,30 @@ func (r *registry) writePB(encoder expfmt.Encoder) error {
 		sort.Sort(metricSorter(mf.Metric))
 	}
 
-	// Write out MetricFamilies sorted by their name.
+	// Write out MetricFamilies sorted by their name, skipping those without
+	// metrics.
```

Comment:
OK, did the research.

The text parser protects against empty metric families here: https://github.com/prometheus/common/blob/master/expfmt/text_parse.go#L105-L110

So this is an issue beyond `client_golang`. I have filed https://github.com/prometheus/common/issues/50 for it.